compiler/lib/lambda_lifting.ml

```diff
@@ -198,12 +198,6 @@ let rec traverse var_depth (program, functions) pc depth limit =
                   Let (f'', Closure (List.map s ~f:snd, (pc'', []))) :: functions
                 in
                 let rem', st = rewrite_body false (program, functions) rem in
-                assert (
```

Comment:
Why is this assertion removed ?

Reply:
The assumption that we never return a function (that we only perform tail calls) does not hold anymore.
And it is no longer an issue to insert a function call right before a `Return` since we now maintain an explicit list of CPS calls.